pom.xml

```diff
@@ -0,0 +1,128 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
@dseapy I kind of thinking, instead of asking the user to build java artifact. Can we add CI action to build and push the artifact in the public repo and version should be associated with numaflow version

Reply:
yeah, that would be much better.  I'm not sure what it takes to get something published to maven central.  i see argo workflows publishes to github packages (https://github.com/argoproj/argo-workflows/blob/b44981f174acebac239e4ecbdda845aa4338f5af/sdks/java/README.md), is that working out ok & should things here be modeled after what's being done over there?

Reply:
i'm happy to help setup whatever makes sense.  i may not have permissions to setup everything, not sure, but I can certainly work towards whatever makes sense & reach out if i get stuck due to permissions.

Reply:
yes

Reply:
We can use maven repository available through GitHub. we can tackle it via a follow-up PR.

Reply:
A couple additional things that might be good to do, which aren't done yet:
* Implement `flatmap` example & put up PR against main numaflow repo for e2e test (https://github.com/numaproj/numaflow/tree/main/test/sdks-e2e#sdks-test)
* Implement a java client with unit tests & examples, so that users can test out their server locally without needing to deploying to k8s or know golang. If doing this, might make sense to have a `common`, `server`, and `client` modules (client & server depending on common) instead of the single one there is now.

If those make sense to do, maybe those as well can be done as follow-up PR?  Regardless of which PR, if those changes are desired, I'd be happy to work on those in my spare time.

Reply:
Yes, we can implement those in follow-up PRs. The java client with unit tests & examples will be awesome to have.